### DISCUSSION TITLE
Developing a scaler

### DISCUSSION THREAD

**Discussion by davidguedalia1** [kedacore/keda#2183] [Q&A / Need Help]
Hi,

I'm trying to deploy a new scaler - I followed this link - https://github.com/kedacore/keda/blob/main/CREATE-NEW-SCALER.md
And when I run "make build" - I got this error:
_make: *** No rule to make target '/scale/interfaces.go', needed by 'pkg/mock/mock_scale/mock_interfaces.go'.  Stop._
Even when I try to run it without the new scaler - with the original Keda code - I got the same error.
Thank you so much

**Comment by tomkerkhove** [comment]
Can you share some thoughts on this @zroubalik ?

**Reply by zroubalik** [reply]
Seems like this is the problematic part: https://github.com/kedacore/keda/blob/7e08fc6bb07b612af32584c322a77447503bfce1/Makefile#L133

It is not able to list your dependencies because you don't have them locally, try to run `go mod tidy` before running `make build` and it should be ok then.

**Reply by davidguedalia1** [reply]
Thank you for helping.
It has a new new Error now - (by running the orginal Keda code) After run make build:
_... keda/bin/mockgen -destination=pkg/mock/mock_scale/mock_interfaces.go -package=mock_scale -source=/cs/usr/.../go/pkg/mod/k8s.io/client-go@v0.22.1/scale/interfaces.go
bash: /cs/usr/.../bin/mockgen: Permission denied
make: *** [Makefile:134: pkg/mock/mock_scale/mock_interfaces.go] Error 126_

Thank you

**Reply by omrifeldman** [reply]
I had the same problem.. someone knows how to fix it?
